frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-object-with-type-property-wrapper.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable lines-between-class-members */
+import { getName, hasLabel, getLabels } from '@console/shared/src';
+import { apiVersionForModel, K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { K8sResourceKindMethods } from '../types/types';
+import { ObjectWithTypePropertyWrapper } from './object-with-type-property-wrapper';
+import { ObjectEnum } from '../../../constants';
+
+export class K8sResourceObjectWithTypePropertyWrapper<
+  RESOURCE extends K8sResourceKind,
+  TYPE extends ObjectEnum<string>,
+  COMBINED_TYPE_DATA,
+  SELF extends K8sResourceObjectWithTypePropertyWrapper<RESOURCE, TYPE, COMBINED_TYPE_DATA, SELF>
+> extends ObjectWithTypePropertyWrapper<RESOURCE, TYPE, COMBINED_TYPE_DATA, SELF>
+  implements K8sResourceKindMethods {
+  getName = () => getName(this.data);
+  getLabels = (defaultValue = {}) => getLabels(this.data, defaultValue);
+  hasLabel = (label: string) => hasLabel(this.data, label);
+
+  setModel = (model: K8sKind) => {
+    this.data.kind = model.kind;
+    this.data.apiVersion = apiVersionForModel(model);
+    return (this as any) as SELF;
```

Comment:
I'm assuming that simply `return this as SELF;` throws typing errors? (`as` should be a force-cast anyway)

Reply:
yes, this one throws a type error